src/System.Net.Primitives/tests/FunctionalTests/IPAddressParsing.cs

```diff
@@ -79,13 +236,136 @@ public void ParseIPv6_WithSubnet_Failure()
         {
             Assert.Throws<FormatException>(() => { IPAddress.Parse("Fe08::/64"); });
         }
+        [Theory]
+        [InlineData("Fe08::1%13542", "fe08::1%13542")]
+        [InlineData("1::%1", "1::%1")]
+        [InlineData("::1%12", "::1%12")]
+        [InlineData("::%123", "::%123")]
+        public void ParseIPv6_ScopeID_Success(string address, string expected)
+        {
+            Assert.Equal(expected, IPAddress.Parse(address).ToString());
+        }
+
+        [Theory]
+        [InlineData("FE08::192.168.0.1", "fe08::c0a8:1")] // Output is not IPv4 mapped
+        [InlineData("::192.168.0.1", "::192.168.0.1")]
+        [InlineData("::FFFF:192.168.0.1", "::ffff:192.168.0.1")] // SIIT
+        public void ParseIPv6_v4_Success(string address, string expected)
+        {
+            Assert.Equal(expected, IPAddress.Parse(address).ToString());
+        }
+
+        [Theory]
+        [PlatformSpecific(~PlatformID.AnyUnix)]
+        // Linux/OSX don't do the IPv6->IPv4 formatting for these addresses
```

Comment:
Should we have a corresponding ```[PlatformSpecific(PlatformID.AnyUnix)]``` that contains the expected value for these platforms?